### ISSUE TITLE
`MD024`: add option to only apply to link anchors

### ISSUE THREAD

**Issue by landure** [rvben/rumdl#581]
### Problem or motivation

`MD024` checks for duplicate headings but ignore custom anchors (for [Hugo](https://gohugo.io/)).

The example markdown below triggers the M024 rule.

### Proposed solution

Add a `allow-different-link-anchors` option:

```toml
[MD024]
allow-different-link-anchors = false  # Allow duplicates at with different link anchors (default: false)
```

### Feature category

Improve existing rule

### Example markdown

```markdown
#### Unit testing

#### Unit testing {#custom-anchor}
```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

### Additional context

`markdownlint` includes the custom anchor in the unique heading check. `allow-different-link-anchors` should default to `true` to match `markdownlint` behavior.

**Comment by rvben** [comment]
Implemented in [v0.1.79](https://github.com/rvben/rumdl/releases/tag/v0.1.79). MD024 now has an `allow_different_link_anchors` option (default `true`) that treats headings with different `{#custom-id}` anchors as distinct. Thanks for the suggestion, @landure!
